### DISCUSSION TITLE
Module federation duplicate common files

### DISCUSSION THREAD

**Discussion by fardad-dev** [callstack/repack#840] [Q&A]
I have a module federation webpack

there are two exposes
a: 'a.js'
b:'b.js'
b is use by a.js too and after creating bundle b is duplicated in a.bundle and b.bundle

how this could be happened and how can I prevent it

**Comment by jbroma** [comment]
Hmm I'll need to verify this - exposing modules like this should result in 1 container bundle and then `a` and `b` chunks. It might be an MF2 issue so I guess it's also worth asking there if that's the expected behaviour.

**Reply by fardad-dev** [reply]
Expected do not duplicate code and use a single version of b. 
This is happening on MF1, I didn't test MF2

**Reply by jbroma** [reply]
can you test MF2? It can have optimizations like that one 👍

**Reply by fardad-dev** [reply]
It is not easy because I should first move from v4 to v5 and then migrate to rspack and then MF2
However, I will try to do that.
